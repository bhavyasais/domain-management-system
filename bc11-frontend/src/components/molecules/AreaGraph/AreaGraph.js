import { Grid, Typography, withStyles } from '@material-ui/core';
import React from 'react';
import { ResponsiveContainer, AreaChart, Area, XAxis, YAxis, CartesianGrid } from 'recharts';
import PropTypes from 'prop-types';
import Dropdown from '../Dropdown/Dropdown';

const styles = () => ({
  container: {
    border: '1px solid  #eeeeee',
    padding: '24px',
    maxWidth: '769px',
  },
  header: {
    height: '40px',
  },
  textColor: {
    color: '#3e3f42',
  },
  headerContainer: {
    marginBottom: '24px',
  },
  axisStyle: {
    fontSize: '10px',
    color: '#949494',
  },
  dot: {
    width: '8px',
    height: '8px',
    borderRadius: '50%',
    // margin: '4px',
  },
});

const AreaGraph = ({ classes, header, list, selectedValue, onDropdownChange }) => {
  return (
    <Grid container className={classes.container}>
      <Grid container item direction="row" className={classes.headerContainer}>
        <Grid item container className={classes.header} xs={6} alignItems="flex-end">
          <Typography className={classes.textColor}>{header}</Typography>
        </Grid>
        <Grid item container xs={6} justify="flex-end">
          <Dropdown selectedItem={selectedValue} list={list} onChange={onDropdownChange} />
        </Grid>
      </Grid>
      {selectedValue?.data && (
        <Grid item xs={12}>
          <ResponsiveContainer width="100%" height={250}>
            <AreaChart data={selectedValue?.data} maxBarSize={50}>
              <CartesianGrid vertical={false} />
              <XAxis
                dataKey="date"
                axisLine={false}
                tickLine={false}
                className={classes.axisStyle}
                padding={{ left: 20 }}
              />
              <YAxis axisLine={false} tickLine={false} className={classes.axisStyle} />
              <Area
                dataKey="value"
                fill={selectedValue.color}
                fillOpacity={0.1}
                stroke={selectedValue.color}
                strokeWidth={2}
                dot
              />
            </AreaChart>
          </ResponsiveContainer>
        </Grid>
      )}
    </Grid>
  );
};

AreaGraph.propTypes = {
  header: PropTypes.string,
  list: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      name: PropTypes.string,
      data: PropTypes.arrayOf(
        PropTypes.shape({
          date: PropTypes.string,
          value: PropTypes.string,
        }),
      ),
    }),
  ),
  selectedValue: PropTypes.shape({
    value: PropTypes.string,
    name: PropTypes.string,
    data: PropTypes.arrayOf(
      PropTypes.shape({
        date: PropTypes.string,
        value: PropTypes.string,
      }),
    ),
  }),
  onDropdownChange: PropTypes.func,
};
AreaGraph.defaultProps = {
  header: 'Default Header',
  list: [],
  selectedValue: { value: '', data: null, name: '' },
  onDropdownChange: null,
};

export default withStyles(styles)(AreaGraph);
