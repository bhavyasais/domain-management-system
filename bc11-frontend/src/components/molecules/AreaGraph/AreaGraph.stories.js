import React, { useState } from 'react';
import AreaGraph from './AreaGraph';

export default {
  title: 'Molecules/AreaGraph',
};
const header = 'Trustscore Trend wrt Relationship Type';
const list = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
    ],
    color: '#ff3162',
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: 'Mon', value: 30 },
      { date: 'Tue', value: 15 },
      { date: 'Wen', value: 12 },
      { date: 'Thu', value: 40 },
      { date: 'Fri', value: 20 },
      { date: 'Sat', value: 5 },
      { date: 'Sun', value: 30 },
    ],
    color: '#ff3162',
  },
];

export const areaGraph = () => {
  const [item, setItem] = useState({
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: 'Mon', value: 30 },
      { date: 'Tue', value: 15 },
      { date: 'Wen', value: 12 },
      { date: 'Thu', value: 40 },
      { date: 'Fri', value: 20 },
      { date: 'Sat', value: 5 },
      { date: 'Sun', value: 30 },
    ],
    color: '#ff3162',
  });
  return (
    <AreaGraph
      header={header}
      list={list}
      selectedValue={item}
      onDropdownChange={(value) => {
        setItem(list.find((obj) => obj.value === value));
      }}
    />
  );
};
