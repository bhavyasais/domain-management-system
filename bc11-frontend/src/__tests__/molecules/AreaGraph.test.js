import React from 'react';
import renderer from 'react-test-renderer';
import AreaGraph from '../../components/molecules/AreaGraph/AreaGraph';

function createNodeMock() {
  const doc = document.implementation.createHTMLDocument();
  return { parentElement: doc.body };
}

describe('<SideNavigationBar/>', () => {
  let mounted;
  let onClick;
  const header = 'Trustscore Trend wrt Relationship Type';
  const list = [
    {
      value: 'zemoso',
      name: 'zemoso',
      data: [
        { date: '1st Dec', value: 30 },
        { date: '2nd Dec', value: 5 },
        { date: '3rd Dec', value: 20 },
        { date: '4th Dec', value: 40 },
        { date: '5th Dec', value: 12 },
        { date: '6th Dec', value: 25 },
        { date: '7th Dec', value: 30 },
      ],
      color: '#ff3162',
    },
    {
      value: 'stanford',
      name: 'stanford',
      data: [
        { date: 'Mon', value: 30 },
        { date: 'Tue', value: 15 },
        { date: 'Wen', value: 12 },
        { date: 'Thu', value: 40 },
        { date: 'Fri', value: 20 },
        { date: 'Sat', value: 5 },
        { date: 'Sun', value: 30 },
      ],
      color: '#ff3162',
    },
  ];

  it('Renders correctly', () => {
    const wrapper = renderer.create(
      <AreaGraph
        header={header}
        list={list}
        onDropdownChange={onClick}
        selectedValue={{
          value: 'stanford',
          name: 'stanford',
          data: [
            { date: 'Mon', value: 30 },
            { date: 'Tue', value: 15 },
            { date: 'Wen', value: 12 },
            { date: 'Thu', value: 40 },
            { date: 'Fri', value: 20 },
            { date: 'Sat', value: 5 },
            { date: 'Sun', value: 30 },
          ],
          color: '#ff3162',
        }}
      />,
      { createNodeMock },
    );
    expect(wrapper).toMatchSnapshot();
  });
});
