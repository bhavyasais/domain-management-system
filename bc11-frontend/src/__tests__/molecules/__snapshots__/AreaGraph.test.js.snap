// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavigationBar/> Renders correctly 1`] = `
<div
  className="MuiGrid-root AreaGraph-container-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root AreaGraph-headerContainer-4 MuiGrid-container MuiGrid-item"
  >
    <div
      className="MuiGrid-root AreaGraph-header-2 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-flex-end MuiGrid-grid-xs-6"
    >
      <p
        className="MuiTypography-root AreaGraph-textColor-3 MuiTypography-body1"
      >
        Trustscore Trend wrt Relationship Type
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-6"
    >
      <div
        className={null}
      >
        <div
          className="MuiFormControl-root Dropdown-formControl-7"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root Dropdown-dropdown-9 MuiInputBase-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="simple-select-label simple-select"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="simple-select"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              stanford
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="stanford"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-11"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="recharts-responsive-container"
      style={
        Object {
          "height": 250,
          "maxHeight": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "none",
            "height": 0,
            "position": "absolute",
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
